### ISSUE TITLE
Improve `savePDF` command

### ISSUE THREAD

**Issue by christian-bromann** [webdriverio/webdriverio#6129]
Two notes:

- we should remove this sentence because it is not true for the pdf feature (Be aware that some browser drivers take screenshots of the whole document (e.g. Geckodriver with Firefox) and others only of the current viewport (e.g. Chromedriver with Chrome).)
- we should allow the options available in printPDF as second parameter

**Comment by NickR23** [comment]
@christian-bromann I ended up adding a new interface for the options, but I'm having trouble with expanding coverage for this new type. 
I added the following to tests/typings/sync/sync.ts but the coverage does not increase
```
// printPage

const pdfPrintOptions: WebdriverIO.PDFPrintOptions = {
    orientation: 'portrait',
    scale: 1,
    background: true,
    width: 24.5,
    height: 26.9,
    top: 10,
    bottom: 10,
    left: 5,
    right: 5,
    shrinkToFit: true,
}
browser.savePDF('./packages/bar.pdf', pdfPrintOptions)
```
- Do you think adding the new type was the right way to go?
- I see the documentation for "Testing Type Definitions", but I'm having trouble carrying this out. Any tips?

Also, apologies for taking so long on this issue! I'm still figuring out the inner workings of the project :)

**Comment by christian-bromann** [comment]
Please add typings similar to this PR: https://github.com/webdriverio/webdriverio/pull/5037/files, meaning you have one main option and then its suboptions, e.g. `getWindowSize`:

```js
 * @param {String}  url      website URL to open
 * @param {NewWindowOptions=} options                newWindow command options
 * @param {String=}           options.windowName     name of the new window
 * @param {String=}           options.windowFeatures features of opened window (e.g. size, position, scrollbars, etc.)
```

`NewWindowOptions` is the main option which needs to be added to `scripts/type-generation/constants.js` and defined in `scripts/templates/webdriverio.tpl.d.ts` (e.g. [here](https://github.com/webdriverio/webdriverio/blob/master/scripts/templates/webdriverio.tpl.d.ts#L465-L468)). Then if you run `npm run generate:typings` it will do the rest for you.

No apologies needed. You are doing a great job. We could definitely improve this process and document it better. Luckily with the TypeScript rewrite it will get to that. Some day.

Let me know if you have further questions!

**Comment by christian-bromann** [comment]
fixed by #6133
